shared/src/main/java/com/faforever/neroxis/mask/Mask.java

```diff
@@ -1067,6 +1067,120 @@ protected U fillCoordinates(Collection<Vector2> coordinates, T value) {
         return (U) this;
     }
 
+    public U fillQuadrilateral(int x1, int y1,
+                               int x2, int y2,
+                               int x3, int y3,
+                               int x4, int y4,
+                               T value) {
+
+        return enqueue(() -> {
+            // First triangle: (1, 2, 3)
+            fillTriangle(x1, y1, x2, y2, x3, y3, value);
+
+            // Second triangle: (1, 3, 4)
+            fillTriangle(x1, y1, x3, y3, x4, y4, value);
+        });
+    }
+
+    public U fillTriangle(int x1, int y1,
+                             int x2, int y2,
+                             int x3, int y3,
+                             T value) {
+
+        return enqueue(() -> {
+            // Copy parameters to local working variables
+            int ax = x1, ay = y1;
+            int bx = x2, by = y2;
+            int cx = x3, cy = y3;
+
+            // Sort vertices by Y (ay <= by <= cy)
+            if (ay > by) {
+                int t = ax; ax = bx; bx = t;
+                t = ay; ay = by; by = t;
+            }
+            if (by > cy) {
+                int t = bx; bx = cx; cx = t;
+                t = by; by = cy; cy = t;
+            }
+            if (ay > by) {
+                int t = ax; ax = bx; bx = t;
+                t = ay; ay = by; by = t;
+            }
+
+            // Handle degenerate case: all vertices on same Y
+            if (ay == cy) {
+                // Degenerate triangle (horizontal line or point) - just draw a line
+                int minX = StrictMath.min(ax, StrictMath.min(bx, cx));
```

Comment:
A little more readable perhaps. Same below
```suggestion
                int minX = IntStream.of(ax, bx, cx).min().orElseThrow();
```

Reply:
I've updated this, it's an interesting technique... pretty neet.
I was gonna say "What about StrictMath? Isn't that important?" but then I realized that this is integers so it's not needed.